lib/Frontend/ParseableInterfaceModuleLoader.cpp

```diff
@@ -582,6 +582,7 @@ class swift::ParseableInterfaceBuilder {
       std::string OutPathStr = OutPath;
       SerializationOpts.OutputPath = OutPathStr.c_str();
       SerializationOpts.ModuleLinkName = FEOpts.ModuleLinkName;
+      SerializationOpts.ParseableInterface = InPath;
```

Comment:
Ah, hm. We can't set this unconditionally because that would expose paths on the build machine when using this for prebuilding. (Or we could make it optionally an SDK-relative path, if you were okay with that, Adrian.)

I think I like the *idea* of an SDK-relative path, but in practice it might be easier to have the manual and caching modes of PIML just be slightly different here.

Reply:
What do you want to do in the prebuild case? Should the SDK dependency still get slurped into the debug info?

Reply:
I'm not up to date on the terminology here. What does prebuilding mean?

When the build machine ≠ link machine, any absolute path in the serialized AST with get us into trouble, including the FILE_DEPENDENCY paths — either we won't be able to locate the files or (only slightly less bad) the builds aren't reproducible.

My plan was that dsymutil would not collect any interfaces that are part of the SDK, since they can always be found in the SDK on the host machine. We could actually enforce this in the compiler by not even setting the parseable interface field for modules that are part of the SDK. Is the SDK-relative bit the right bit to check for this? Or are SDK modules built in a multi-step process where this bit wouldn't be set?

That still doesn't solve the problem of having absolute paths (any of them) in the .swiftmodule and thus in the debug info. I also want to solve this, but since it is also affect other fields, such as the dependencies, I'd like to do that separately. For example, we introduce a `--build-root` option that all paths are relative to. But let's save that for later.